back-end/dao/projectsDAO.js

```diff
@@ -1,71 +1,146 @@
-     * @returns an array of all the projects from the database
+     * Create a project object in MongoDB (see AddProject.js in /front-end for more info).
+     * @param {Object} body A project object that has project data
+     * @returns A list of project objects
      */
-    static async getFilteredProjects({
-         filters = null
-    } = {})
-    {
-        let query
-        if (filters) {
-            if ("project_name" in filters) {
-                query = { "project_name": { $eq: filters["project_name"] } }
-            } else if ("project_id" in filters) {
-                query = { "project_id": { $eq: filters["project_id"] } }
-            }
-        }
-        let cursor
+    async createProject(body) {
         try {
-            cursor = await projects
-                .find(query)
-            return cursor.toArray()
+            await Projects.create(body);
+            return {
+                status: 200,
+                data: {response: `successfully created project`}
+            }
         } catch (e) {
-            console.error(`Unable to issue find command, ${e}`)
-            return []
+            console.error(`Unable to issue create command, ${e}`);
+            return {
+                status: 500,
+                data: { error: e.message },
+            };
         }
     }
 
+
     /**
-     * Create a project object in MongoDB (see AddProject.js in /front-end for more info).
+     * Delete the project object with the name projectName from the database.
+     * @param {String} projectName Name of the project to be deleted
+     * @returns {Promise<{error}|*>}
      */
-    static async createProject(req, res, next){
-        await projects.insertOne(req)
-    }
-
-    static async deleteProject(projectName) {
+    async deleteProject(projectName) {
         try {
-            const deleteResponse = await projects.deleteOne({
-                project_name: projectName
-            })
-            return deleteResponse
+            await Projects.deleteOne({project_name: projectName});
+            return {
+                status: 200,
+                data: { response: `successfully deleted project ${projectName}` },
+            };
         } catch (e) {
-            console.error(`Unable to delete project: ${e}`)
-            return { error: e }
+            console.error(`Unable to issue delete command, ${e}`);
+            return {
+                status: 500,
+                data: { error: e.message },
+            };
         }
     }
 
+
     /**
-     * Get an array of all project objects from MongoDB.
+     * Get a project object with a particular id from MongoDB.
+     * Also, add parsed and trimmed transcripts corresponding to that project to the project object.
+     * @param id
+     * @returns {Promise<*>}
      */
-    static async getAllProjects() {
-        return await projects.find().toArray();
+    async getProjectByID(id) {
+        try {
+            const pipeline = [
```

Comment:
What does this pipeline do exactly?

Reply:
This is one of Khushi's implementations, but I believe it aggregates and modifies documents in the specific stages you want; in this case, it's used to aggregate cleaned and test transcripts with a given project id. (we should probably document this lol)